core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -230,6 +226,22 @@ private void closeUfsBlock(long sessionId, long blockId, boolean successful)
     }
   }
 
+  private void abortUfsBlock(long sessionId, long blockId) throws IOException {
+    mUnderFileSystemBlockStore.closeBlock(sessionId, blockId);
+    Optional<TempBlockMeta> tempBlockMeta = mLocalBlockStore.getTempBlockMeta(blockId);
+    if (tempBlockMeta.isPresent() && tempBlockMeta.get().getSessionId() == sessionId) {
```

Comment:
shall we have a log warn if this condition does not meet?

Reply:
good call, just added